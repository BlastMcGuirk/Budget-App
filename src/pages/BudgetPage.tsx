import React from 'react';
import { View, StyleSheet } from 'react-native';
import BudgetSummary from '../components/BudgetSummary';

export default function BudgetPage() {
    return (
        <View style={styles.budgetSummaries}>
            <BudgetSummary
                budget='needs'
                remaining={1234.56}
                total={1500}/>
            <BudgetSummary 
                budget='wants'
                remaining={-123.45}
                total={500}/>
        </View>
    )
}

const styles = StyleSheet.create({
    budgetSummaries: {
      flexDirection: 'row',
    },
  });
  