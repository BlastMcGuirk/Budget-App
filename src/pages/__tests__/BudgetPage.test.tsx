import renderer from 'react-test-renderer';
import BudgetPage from "../BudgetPage";

describe("Budget Page Tests", () => {

    it("Renders correctly", () => {
        const tree = renderer.
            create(<BudgetPage />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

});