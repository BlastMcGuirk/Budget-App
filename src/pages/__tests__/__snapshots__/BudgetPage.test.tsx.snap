// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Budget Page Tests Renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "padding": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 22,
        }
      }
    >
      NEEDS
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontSize": 36,
          },
          Object {
            "color": "green",
          },
        ]
      }
    >
      $
      1234.56
    </Text>
    <Text
      style={
        Object {
          "fontSize": 16,
        }
      }
    >
      / $
      1500
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "padding": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 22,
        }
      }
    >
      WANTS
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontSize": 36,
          },
          Object {
            "color": "red",
          },
        ]
      }
    >
      $
      -123.45
    </Text>
    <Text
      style={
        Object {
          "fontSize": 16,
        }
      }
    >
      / $
      500
    </Text>
  </View>
</View>
`;
